test/libsolidity/ASTJSONTest.cpp

```diff
@@ -131,93 +131,106 @@ TestCase::TestResult ASTJSONTest::run(ostream& _stream, string const& _linePrefi
 	}
 	c.setSources(sources);
 	c.setEVMVersion(solidity::test::CommonOptions::get().evmVersion());
-	if (c.parse())
-		c.analyze();
-	else
+
+	if (!c.parse())
 	{
 		SourceReferenceFormatterHuman formatter(_stream, _formatted, false);
 		for (auto const& error: c.errors())
 			formatter.printErrorInformation(*error);
 		return TestResult::FatalError;
 	}
 
+	c.analyze();
+
+	bool resultsMatch = runTest(
+		m_expectation,
+		m_result,
+		sourceIndices,
+		c,
+		false,
+		"",
+		_stream,
+		_linePrefix,
+		_formatted
+	);
+
+	resultsMatch = runTest(
+		m_expectationLegacy,
+		m_resultLegacy,
+		sourceIndices,
+		c,
+		true,
+		"legacy",
+		_stream,
+		_linePrefix,
+		_formatted
+	) && resultsMatch;
+
+	return resultsMatch ? TestResult::Success : TestResult::Failure;
+}
+
+bool ASTJSONTest::runTest(
+	string& _expectation,
+	string& _result,
+	map<string, unsigned> const& _sourceIndices,
+	CompilerStack& _compiler,
+	bool _legacy,
+	string const& _variation,
```

Comment:
Maybe this parameter can be calculated based on other parameters, so it can be removed?

Reply:
See my other response